### ISSUE TITLE
[XLA] RET_CHECK abort compiling reduce_window with an all-1 window and only negative padding

### ISSUE THREAD

**Issue by EylonKrause** [openxla/xla#48719]
Compiling a `reduce_window` whose window is all-1 (size 1, stride 1, no dilation) and whose window padding is **negative on every side that has any padding at all** aborts the process with a `RET_CHECK` failure instead of compiling.

Negative window padding on `reduce_window` is valid and JAX emits it, so this takes down the process on a legitimate program.

### Reproducer

```python
import os
os.environ["JAX_PLATFORMS"] = "cpu"
import numpy as np, jax, jax.numpy as jnp
from jax import lax

x = jnp.asarray(np.arange(32).reshape(8, 4), jnp.float32)
f = lambda a: lax.reduce_window(a, 0.0, lax.add, (1, 1), (1, 1), ((-1, 0), (0, 0)))
print(np.asarray(jax.jit(f)(x)).shape)   # expected (7, 4)
```

```
E0911 16:04:05.227864 status_macros.cc:58] INTERNAL: RET_CHECK failure
(external/xla+/xla/hlo/ir/hlo_computation.cc:1773)
ShapeUtil::Compatible(old_instruction->shape(), new_instruction->shape())
"reduce_window_sum.7" (f32[7,4]{1,0}) vs "add" (f32[8,4]{1,0})
=== Source Location Trace: ===
external/xla+/xla/status_macros.cc:81
```

The process dies with `SIGABRT` (exit 134). Eager execution of the same op is fine.

### Which padding crashes

Same input, window `(1, 1)`, stride `(1, 1)`:

| padding | result |
|---|---|
| `((-1, 0), (0, 0))` | **abort** |
| `((0, -1), (0, 0))` | **abort** |
| `((-1, -1), (0, 0))` | **abort** |
| `((-2, 0),)` (rank 1) | **abort** |
| `((-2, 1), (0, 0))` | ok, shape `(7, 4)` |
| `((-3, 1),)` (rank 1) | ok, shape `(6,)` |
| `((-2, 1),)` with window 2 | ok |

So it aborts exactly when the window is all-1 **and** no padding amount is positive.

### Analysis

The shapes in the message say it directly: a `reduce_window` producing `f32[7,4]` is being replaced by its operand `add`, which is `f32[8,4]`.

`AlgebraicSimplifier::HandleReduceWindow` has an early rewrite for a window that selects a single element: such a `reduce_window` is the identity on its operand, optionally wrapped in a `pad` for the window padding. That wrapper is only built when some padding amount is positive, so with purely negative (cropping) padding the operand is substituted directly, its shape still carries the uncropped dimension, and `ReplaceInstruction`'s `ShapeUtil::Compatible` check fires.

Negative window padding needs the mirror of the positive case here (a slice rather than a pad), or the rewrite should bail out when any padding amount is negative.

### Environment

- `jax` 0.11.1, `jaxlib` 0.11.1 (pip), CPU backend, `JAX_PLATFORMS=cpu`
- Linux x86-64 (AVX-512), Python 3.12

### Notes

Found by a differential fuzzer over `reduce_window` parameters and minimised by hand; every row of the table above was re-run in a fresh process. I am filing this as an issue rather than a pull request, per the repository's contribution guidelines: this report was prepared with the help of an AI coding assistant (Claude), and I reproduced every result above myself on the environment listed.

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/982938038.
